syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";


message MongoDB {
  message MongoQuery {
    google.protobuf.StringValue database = 1;
    google.protobuf.StringValue collection = 2;
    google.protobuf.StringValue order_by_field = 3;
    google.protobuf.UInt64Value limit = 4;
    google.protobuf.StringValue filters = 5;
    google.protobuf.StringValue projection = 6;
    google.protobuf.UInt64Value timeout = 7;
  }

  enum TaskType {
    UNKNOWN = 0;
    DATA_FETCH = 1;
  }

  TaskType type = 1;
  oneof task {
    MongoQuery data_fetch = 2;
  }
}